lib/connection_pool.rb

```diff
@@ -57,11 +57,7 @@ def self.after_fork
         # All we need to do is to ensure the main thread doesn't have a
         # checked out connection
         pool.checkin(force: true)
-        pool.reload do |connection|
-          # Unfortunately we don't know what method to call to close the connection,
-          # so we try the most common one.
-          connection.close if connection.respond_to?(:close)
-        end
+        pool.reload
```

Comment:
I don't follow how this fixes the problem?

Most IOs will have a finalizer to call `close` or equivalent when they are GCed, so this only delays the problem.

Reply:
Assuming it is one of those IO's and it doesn't implement its own fork-safety, that's true, it won't solve the issue, but that issue can be solved in the client library instead.

To bias toward safety since client libraries might not all handle this, we could also drain and close _before_ fork so the parent state is also clean; it prevents clients from optimally responding (and just disconnecting the child) but they can just disable `auto_reload_after_fork` on those pools if they want.